pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -50,7 +50,18 @@ func (o *ClusterUninstaller) listCloudControllerTargetPools() ([]cloudResource,
 		for _, instanceURL := range pool.Instances {
 			name, _ := o.getInstanceNameAndZone(instanceURL)
```

Comment:
This approach seems reasonable, but I'm concerned we're complicating the existing approach when we might be able to solve the problem by simplifying it. The existing approach is to find all target pools that match the cloud controller name convention, and then iterate through all of the instances in those target pool; if all of those instances belong to the cluster we mark the target pool for delete.

We could have this approach for two reasons I see: 
1. We don't know how to identify cluster target pools
2. We don't want to delete a target pool if it has a non-cluster instance

For question 1, is it possible to identify the target pools by using `loadBalancerName`? Spot checking GCP it looks like the target pool matches the `loadBalancerName` exactly. 

For question 2, I find it unlikely that a non cluster VM would be added to the cluster load balancer, and moreoever what happens if we try to destroy a target pool with a non-cluster vm? We should test this. 

So my suggestion for an alternative approach would be to change the filter to match only on `loadBalancerName` and remove the instance checks entirely.

Reply:
I might be a bit confused on this one, because the loadBalancerName is assumed to be the pool.Name. How can we use the loadBalancerName if we haven't found the target pool yet?